src/test/java/uk/gov/justice/digital/hmpps/vsip/pages/SelectTimeSlotPage.java

```diff
@@ -50,20 +50,10 @@ public void clickNextEnabledElement() {
         methodsService.click("xpath", "//button[contains(@class, 'govuk-accordion__show-all')]");
 
         // List of different XPaths for time slots elements
-        List<String> xpaths = new ArrayList<>();
-        xpaths.add("//*[@id='1']");
-        xpaths.add("//*[@id='2']");
-        xpaths.add("//*[@id='3']");
-        xpaths.add("//*[@id='4']");
-        xpaths.add("//*[@id='5']");
-        xpaths.add("//*[@id='6']");
-        xpaths.add("//*[@id='7']");
-        xpaths.add("//*[@id='8']");
-        xpaths.add("//*[@id='9']");
-        xpaths.add("//*[@id='10']");
-
-        // Iterate through the XPaths
-        for (String xpath : xpaths) {
+        int numberOfElements = 30;  // or any other number of elements you have
```

Comment:
I don't know about XPath, but with CSS selectors you can target the first enabled radio button in one go (`"input[name=visit-date-and-time]:not([disabled]):first"`).

So wondering if rather than looping through a fixed number of elements you could just replace this whole block of code with something like:

```
methodsService.click("css", "input[name=visit-date-and-time]:not([disabled]):first")
```

(n.b. untested - just a guess looking at `CommonMethodsService` implementation!)

Reply:
I'll give it a go locally & test it. Thank you